pkg/webhook/admission/webhook.go

```diff
@@ -166,15 +174,38 @@ func (wh *Webhook) Handle(ctx context.Context, req Request) (response Response)
 		}()
 	}
 
+	reqLog := wh.getLogger(&req)
+	reqLog = reqLog.WithValues("requestID", req.UID)
+	ctx = logf.IntoContext(ctx, reqLog)
+
 	resp := wh.Handler.Handle(ctx, req)
 	if err := resp.Complete(req); err != nil {
-		wh.log.Error(err, "unable to encode response")
+		reqLog.Error(err, "unable to encode response")
 		return Errored(http.StatusInternalServerError, errUnableToEncodeResponse)
 	}
 
 	return resp
 }
 
+// getLogger constructs a logger from the injected log and LogConstructor.
```

Comment:
I wonder if we should implement this the same way as we did for the controller:
```go
	if options.LogConstructor == nil {
		log := mgr.GetLogger().WithValues(
			"controller", name,
		)
		options.LogConstructor = func(req *reconcile.Request) logr.Logger {
			log := log
			if req != nil {
				log = log.WithValues(
					"object", klog.KRef(req.Namespace, req.Name),
					"namespace", req.Namespace, "name", req.Name,
				)
			}
			return log
		}
	}
```

The delta is roughly:
* LogConstructor would have this signature: `LogConstructor func(req *Request) logr.Logger`
* The default log constructor would then use the base logger directly
* namespace & name are also added as top-level k/v pairs in addition to the nested values below object

e.g.:
```go
// getLogger constructs a logger from the injected log and LogConstructor.
func (wh *Webhook) getLogger(req *Request) logr.Logger {
	if wh.LogConstructor == nil {
		if req != nil {
			return wh.log.WithValues("object", klog.KRef(req.Namespace, req.Name),
				"namespace", req.Namespace, "name", req.Name,
				"resource", req.Resource, "user", req.UserInfo.Username,
			)
		}
		return wh.log
	}
	return wh.LogConstructor(req)
}
```

Reply:
> LogConstructor would have this signature: `LogConstructor func(req *Request) logr.Logger`

I added the `base logr.Logger` param to the `LogConstructor` to allow using the logger injected by the webhook server which includes the webhook path:
https://github.com/kubernetes-sigs/controller-runtime/blob/869888c3e70c8c87f33e3ca000aaef451c6c5cbb/pkg/webhook/server.go#L165

This is different from controllers, which use the manager's logger directly (which doesn't carry any additional information about the controller):
https://github.com/kubernetes-sigs/controller-runtime/blob/598978c592839b5a602af3706fce8751d2d2532c/pkg/controller/controller.go#L110

With this, one can add additional fields via a custom `LogConstructor` without adding the `webhook` field again.
If one wants to ignore the fields of the base logger, one can simply use any other logger.

I get that this is different from the controller loggers. If harmonization is more important here than preserving the injected fields, I will apply your suggestion. :)
@sbueringer WDYT?

Reply:
Looks good to me. Thank you very much!!